### PR TITLE
Fix many warnings.

### PR DESCRIPTION
**Pull request by JoshDuMan**

### CODE REVIEW

**Comment by BowserSlug** [comment]
✅ This PR fixes 389 warnings!

File: src/world/world.c [hunk 1]
```diff
@@ -221,17 +221,17 @@ void* load_asset_by_name(const char* assetName, u32* decompressedSize) {
...
-    dma_copy(ASSET_TABLE_FIRST_ENTRY, ASSET_TABLE_FIRST_ENTRY + firstHeader.offset, assetTableBuffer);
+    dma_copy((u8* ) ASSET_TABLE_FIRST_ENTRY, (u8* ) ASSET_TABLE_FIRST_ENTRY + firstHeader.offset, assetTableBuffer);
     while (strcmp(curAsset->name, assetName) != 0) {
         curAsset++;
     }
     *decompressedSize = curAsset->decompressedLength;
     ret = general_heap_malloc(curAsset->compressedLength);
-    dma_copy(ASSET_TABLE_FIRST_ENTRY + curAsset->offset,
-             ASSET_TABLE_FIRST_ENTRY + curAsset->offset + curAsset->compressedLength, ret);
+    dma_copy((u8* ) ASSET_TABLE_FIRST_ENTRY + curAsset->offset,
```
Comment (nanaian) [inline src/world/world.c:233]: This is weird, perhaps dma_copy should take a `void*` instead?

Comment (nanaian) [inline src/world/world.c:233]:
```suggestion
    dma_copy((u8*) ASSET_TABLE_FIRST_ENTRY + curAsset->offset,
```

File: src/7E9D0.c [hunk 2]
```diff
@@ -149,7 +148,7 @@ void set_action_state(s32 actionState) {
     }
 
     if (playerStatus->unk_D8) {
-        playerStatus->unk_D8[3][9] = 0xA;
+        playerStatus->unk_D8[3][9] = 0xA; // TODO: Substruct? Currently UB.
```
Comment (nanaian) [inline src/7E9D0.c:151]: pls explain further

Comment (JoshDuMan) [reply src/7E9D0.c:151]: Unk_d8 is an s32*, not an array

File: src/world/partners.c [hunk 3]
```diff
@@ -39,6 +39,9 @@ extern s32 D_8010CFB8;
 extern s32 D_8010CFBC;
 extern s32 D_8010CFCC;
 
+extern s32 D_801083D8[];
+extern s32 D_80108068[];
```
Comment (ethteck) [inline src/world/partners.c:43]: these are out of order - the ones you added should be swapped and come before the CF* ones

**Review by ethteck — APPROVED** [review]

**Review by Brotenko — APPROVED** [review]

**Review by nanaian — APPROVED** [review]

**Pull request merged**